src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -335,9 +335,19 @@ internal void LogBuffer(int size)
         private SocketError FinishOperationAccept(Internals.SocketAddress remoteSocketAddress)
         {
             System.Buffer.BlockCopy(_acceptBuffer!, 0, remoteSocketAddress.Buffer, 0, _acceptAddressBufferCount);
+            // We already assigned a non-null socket to this in GetOrCreateFunc
+            Socket sukru = _acceptSocket!;
             _acceptSocket = _currentSocket!.CreateAcceptSocket(
                 SocketPal.CreateSocket(_acceptedFileDescriptor),
                 _currentSocket._rightEndPoint!.Create(remoteSocketAddress));
+            sukru.DisposeHandle();
```

Comment:
Would it make sense to leave `_acceptSocket` `null` as it is now and do all the operations bellow only if it is set? This change will make this operation more expensive in all cases - while Accept with existing socket should be rare IMHO.

Reply:
Yeah, I think it would. I wanted to make the signature the same as windows, but I'll convert it to the old signature and implement this behavior.